include/nccl_ofi_api.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2017-2022, NVIDIA CORPORATION. All rights reserved.
+ * Copyright (c) 2022-2023 Amazon.com, Inc. or its affiliates. All rights reserved.
+ */
+
+#ifndef NET_OFI_API_H_
+#define NET_OFI_API_H_
+
+#include "nccl-headers/net.h"
+#include "nccl-headers/error.h"
+
+#ifdef _cplusplus
+extern "C" {
+#endif
+
+ncclResult_t nccl_net_ofi_init(ncclDebugLogger_t logFunction);
+ncclResult_t nccl_net_ofi_init_v3(ncclDebugLogger_t logFunction);
+ncclResult_t nccl_net_ofi_devices(int *ndev);
+ncclResult_t nccl_net_ofi_getProperties(int dev, ncclNetProperties_v6_t *props);
+ncclResult_t nccl_net_ofi_pciPath_v2(int dev, char** path);
+ncclResult_t nccl_net_ofi_ptrSupport_v2(int dev, int *supportedTypes);
+ncclResult_t nccl_net_ofi_getProperties_v4(int dev, ncclNetProperties_v4_t* props);
+ncclResult_t nccl_net_ofi_listen(int dev, void *handle, void **listenComm);
+ncclResult_t nccl_net_ofi_listen_v4(int dev, void* handle, void** listenComm);
+ncclResult_t nccl_net_ofi_connect(int dev, void* handle, void** sendComm);
+ncclResult_t nccl_net_ofi_connect_v4(int dev, void* handle, void** sendComm);
```

Comment:
FOr cases like this and a couple others that have variants, wouldn't it be better to have a single API here in nccl_ofi_api.h and have each accelerator-specific code register a wrapper if needed? This would leave the versioning fully out of the new API.

Reply:
Yeah, it's tricky.  I think we can push a little bit more (I'd like to get the `ncclNetProperties_v6_t` structure out of the core, but some of the other changes are going to end up duplicating more code than I'd really like.  So I kind of stopped here, as it unblocks Ryan.